src/dbus/server/dbus_object.cpp

```diff
@@ -252,6 +273,40 @@ void DBusObject::SetPropertyMethodHandler(DBusRequest &aRequest)
     return;
 }
 
+void DBusObject::AsyncGetPropertyMethodHandler(DBusRequest &aRequest)
+{
+    DBusMessageIter iter;
+    std::string     interfaceName;
+    otError         error = OT_ERROR_NONE;
+    std::string     propertyName;
+
+    VerifyOrExit(dbus_message_iter_init(aRequest.GetMessage(), &iter), error = OT_ERROR_FAILED);
+    VerifyOrExit(DBusMessageExtract(&iter, interfaceName) == OTBR_ERROR_NONE, error = OT_ERROR_PARSE);
+    VerifyOrExit(DBusMessageExtract(&iter, propertyName) == OTBR_ERROR_NONE, error = OT_ERROR_PARSE);
+
+    {
+        auto propertyIter = mAsyncGetPropertyHandlers.find(interfaceName);
+
+        otbrLogDebug("AsyncGetProperty %s.%s", interfaceName.c_str(), propertyName.c_str());
+        VerifyOrExit(propertyIter != mAsyncGetPropertyHandlers.end(), error = OT_ERROR_NOT_FOUND);
+        {
+            auto &interfaceHandlers = propertyIter->second;
+            auto  interfaceIter     = interfaceHandlers.find(propertyName);
+
+            VerifyOrExit(interfaceIter != interfaceHandlers.end(), error = OT_ERROR_NOT_FOUND);
+            (interfaceIter->second)(aRequest);
+        }
+    }
+
+exit:
+    if (error != OT_ERROR_NONE)
+    {
+        otbrLogWarning("GetProperty %s.%s error:%s", interfaceName.c_str(), propertyName.c_str(),
+                       ConvertToDBusErrorName(error));
+        aRequest.ReplyOtResult(error);
```

Comment:
This only happens when there's an error. Is it as expected?

Reply:
Yes, this is expected. When there is no error, the request is expected to be called in the handler: `(interfaceIter->second)(aRequest);`